### PR TITLE
Added Purple Path slab

### PR DESCRIPTION
**Pull request by AdamPlenty**

### CODE REVIEW

**Comment by dotted** [comment]
Is this still being worked on?

**Comment by AdamPlenty** [comment]
> 
> 
> Is this still being worked on?

I almost forgot about this! I think the only thing missing is the description, but I don't know how to do that.

**Comment by Loobinex** [comment]
Placing it all seems to work well, so if the slab type is added to KeeperFX, could be merged, see:
https://github.com/dkfans/keeperfx/pull/915

File: libadikted/obj_column.c [hunk 1]
```diff
@@ -249,6 +249,9 @@ void create_columns_for_slab(struct COLUMN_REC *clm_recs[9],struct LEVOPTIONS *o
     case SLAB_TYPE_GUARDPOST:
       create_columns_slb_guardpost(clm_recs,surr_slb,surr_own,surr_tng);
       break;
+    case 54:
```
Comment (dotted) [inline libadikted/obj_column.c:252]: Add `#define SLAB_TYPE_PURPLE_PATH    0x36` in `obj_slabs.h` and use that instead so magic numbers is avoided.

File: mapslang/scr_list.c [hunk 2]
```diff
@@ -682,7 +682,7 @@ void draw_mdcclm(struct SCRMODE_DATA *scrmode,struct WORKMODE_DATA *workdata)
 void draw_mdslbl(struct SCRMODE_DATA *scrmode,struct WORKMODE_DATA *workdata)
 {
     draw_numbered_list(get_slab_fullname,scrmode,workdata,
-        0,SLAB_TYPE_GUARDPOST,17);
```
Comment (dotted) [inline mapslang/scr_list.c:685]: As above don't use magic numbers.

File: mapslang/map.hlp [hunk 3]
```diff
@@ -1004,6 +1004,22 @@ with player's flag on center.
...
+\rPurple path\s
+
+Slab whose appearance is normally
+used when the game cannot process 
+something.
+
+
+
+
+Unknown slab
```
Comment (dotted) [inline mapslang/map.hlp:1016]: Missing `\r` and `\s`?

**Review by dotted — CHANGES_REQUESTED** [review]
Just minor things, looks good otherwise

**Comment by PieterVdc** [comment]
@dotted I implemented the requested changes on this one

**Review by dotted — APPROVED** [review]

**Pull request merged**